### PR TITLE
feat(UI): CLI file clean up assistant

### PR DESCRIPTION
**Pull request by JaideepPalit**
> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*#1365*)
> * Did you add or update any new dependencies that are required for your change? - No

Issue: closes: #1365 

**Refer** - #1365 

 
> Test Attachment Usage/ Attachment Select in Generate Clearing Report/Readme OSS/Source Code Download of Project.
    ```
in project clearing / attachment selection / attachment usage views: this CLI file does not appear anymore
exception only for the projects where it has been actively selected (attachment usage has been set for tis file) - greyed out
    ```

> Have you implemented any additional tests? - No

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>

### CODE REVIEW

**Comment by JaideepPalit** [comment]
Resolved merge conflict

**Review by ravi110336 — COMMENTED** [review]
Code looks good to me, please consider some minor review comments. and one observation which we discussed in our last week call with Michael. 

For eg: If we attach 2 simple CLI files without having any variations and user clicks on the Evaluate CLI Files, we see the text as Evaluating and since there is no variations in data nothing happens(which is expected behavior), so in this scenario may be if we can show some generic message in the screen like "No Matching data found in both the CLI files." 

evaluateCLIFile

**Comment by ravi110336** [comment]
Tested the below use case and working fine.
1. Added the 2 CLI file with the data variants in copyright, License Data and Obligation, it greyed out the CLI file which is part of the another super CLI file attachment.
 
Test CLI File

2. Attachment Usage in Projects tab shows only one CLI file which is not greyed.

attachmentUsage

3. Project Attachment will have both the attachment.
projectAttachment

4. Also tested the Clearing Report, Readme OSS, and Source Code Download of Project.

File: backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java [hunk 1]
```diff
@@ -167,6 +182,361 @@ public LicenseInfoFile getLicenseInfoFile(Project project, User user, String out
...
+
+                    Map<String, Set<String>> secCopyrightsWithFilesHash = secondaryEntry.getValue().getLicenseInfo()
+                            .getCopyrightsWithFilesHash();
+                    Set<LicenseNameWithText> secLicenseNamesWithTexts = secondaryEntry.getValue().getLicenseInfo()
+                            .getLicenseNamesWithTexts();
+                    if (CommonUtils.isNullOrEmptyMap(secCopyrightsWithFilesHash)
+                            && !CommonUtils.isNotEmpty(secLicenseNamesWithTexts)) {
+                        listIteratorSec.remove();
+                        removedEntry.add(secondaryEntry.getKey().getAttachmentContentId());
+                        Map<String, String> metadata = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (JaideepPalit) [reply backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java:394]: In this case , Entry of a map cannot be null.
However please let me know, if you have an example of null Entry in a Map

File: backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java [hunk 2]
```diff
@@ -167,6 +182,361 @@ public LicenseInfoFile getLicenseInfoFile(Project project, User user, String out
...
+        }
+
+    }
+
+    private void groupParsedObligationsByComponentSHA(
+            Map<String, List<Entry<Attachment, ObligationParsingResult>>> groupedParsedObligations,
+            Map<Attachment, ObligationParsingResult> parsedObligations) {
+
+        for (Entry<Attachment, ObligationParsingResult> parsedObligationEntry : parsedObligations.entrySet()) {
+            if (parsedObligationEntry.getValue() != null && parsedObligationEntry.getValue() != null) {
```
[the remark this answers is not in the document]
Comment (JaideepPalit) [reply backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/LicenseInfoHandler.java]: removed duplicate check

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/attachmentsDetail.jsp [hunk 3]
```diff
@@ -186,27 +221,49 @@
...
+                    <table class="table table-borderless"
+                         <tr class="dataTableChildRowCell">
+                             <td><liferay-ui:message key="sha" /> : </td>
+                             <td>`+ rowData.sha1 + `</td>
+                             <td><liferay-ui:message key="uploaded.on" /> : </td>
+                             <td>` + rowData.uploadedOn + `</td>
+                             <td><liferay-ui:message key="uploaded.comment" /> : </td>
+                             <td>` + rowData.uploadedComment + `</td>
+                         <tr>
+                         <tr class="dataTableChildRowCell " ` + className + `">
```
[the remark this answers is not in the document]
Comment (JaideepPalit) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/attachmentsDetail.jsp:250]: removed

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/attachmentsDetail.jsp [hunk 4]
```diff
@@ -186,27 +221,49 @@
...
+                             <td><liferay-ui:message key="checked.on" /> : </td>
+                             <td>` + rowData.checkedOn + `</td>
+                             <td><liferay-ui:message key="checked.comment" /> : </td>
+                             <td>` + rowData.checkedComment + `</td>
+                             <td></td>
+                             <td></td>
+                         <tr>`
+
+                    if(rowData.superAttachmentId) {
```
[the remark this answers is not in the document]
Comment (JaideepPalit) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/utils/includes/attachmentsDetail.jsp:259]: fixed

**Comment by JaideepPalit** [comment]
Thanks @ravi110336 for review and test.
Incorporated changes based on review comments

**Comment by ravi110336** [comment]
Thanks @JaideepPalit , for incorporating the changes the code looks good to me and tested the new functionality which was discussed in our meeting and it is working fine.

Generic message for CLI

new feature CLI

**Pull request merged**